File: LcWithPatterns/JavaScript/Stack/lc383-RansomNote.js
"use strict";
function canConstruct(ransomNote, magazine) {
    const charConstructMagazine = new Map();
    for (const char of magazine) {
        charConstructMagazine.set(char, (charConstructMagazine.get(char) ?? 0) + 1);
    }
    for (const char of ransomNote) {
        const count = charConstructMagazine.get(char) ?? 0;
        if (count === 0) {
            return false;
        }
        charConstructMagazine.set(char, count - 1);
    }
    return true;
}
;

File: LcWithPatterns/Python/Stack/lc383-RansomNote.py
def canConstruct(ransomNote: str, magazine: str) -> bool:
    charConstructMagazine = {}

    for char in magazine:
        charConstructMagazine[char] = charConstructMagazine.get(char, 0) + 1

    for char in ransomNote:
        count = charConstructMagazine.get(char, 0)
        if count == 0:
            return False
        charConstructMagazine[char] = count - 1

    return True
